lisa/core/customScript.py

```diff
@@ -0,0 +1,98 @@
+from __future__ import annotations
+
+from hashlib import md5
+from pathlib import Path
+from typing import TYPE_CHECKING, List, Optional, Type
+
+from lisa.core.tool import Tool
+from lisa.util.executableResult import ExecutableResult
+from lisa.util.process import Process
+
+if TYPE_CHECKING:
+    from lisa.core.node import Node
+
+
+class CustomScript:
+    def __init__(self, node: Node, cwd: Path, command: str) -> None:
+        self.node = node
+        self.cwd = cwd
+        self.command = command
+
+    def runAsync(
+        self,
+        extraParameters: str = "",
+        noErrorLog: bool = False,
+        noInfoLog: bool = False,
+    ) -> Process:
+        command = f"{self.command} {extraParameters}"
+        return self.node.executeAsync(
+            cmd=command,
+            useBash=True,
+            noErrorLog=noErrorLog,
+            noInfoLog=noInfoLog,
+            cwd=self.cwd,
+        )
+
+    def run(
+        self,
+        extraParameters: str = "",
+        noErrorLog: bool = False,
+        noInfoLog: bool = False,
+    ) -> ExecutableResult:
+        process = self.runAsync(
+            extraParameters=extraParameters, noErrorLog=noErrorLog, noInfoLog=noInfoLog,
+        )
+        return process.waitResult()
+
+
+class CustomScriptSpec:
+    def __init__(
+        self,
+        root_path: Path,
+        files: List[str],
+        command: Optional[str] = None,
+        dependencies: Optional[List[Type[Tool]]] = None,
+    ) -> None:
+        if not files:
+            raise Exception("CustomScriptSpec should have at least one file")
+
+        self.files = files
+        self.dependencies = dependencies
+
+        self._localRootPath: Path = root_path
+
+        if command:
+            self.command = command
+        else:
+            self.command = files[0]
+
+        # generate an unique name for dict hash
+        hash_source = "".join(files).encode("utf-8")
+        hash_result = md5(hash_source)
```

Comment:
I might suggest we switch to `sha256` just so one one asks us why we're using `md5`. None of the considerations of MD5 apply here, but it's gonna get asked since this is OSS 😂

Reply:
sure 😃